burr/integrations/persisters/b_mongodb.py

```diff
@@ -0,0 +1,74 @@
+from pymongo import MongoClient
+import datetime
+import json
+import logging
+from typing import Literal, Optional
+from burr.core import persistence, state
+from datetime import datetime, timezone
+
+logger = logging.getLogger(__name__)
+
+class MongoDBPersister(persistence.BaseStatePersister):
+    """A class used to represent a MongoDB Persister."""
+
+    def __init__(self, uri='mongodb://localhost:27017', db_name='mydatabase', collection_name='mystates'):
```

Comment:
question: does the mongo URI encode things like username & password? If so we'll want some convenience functions like the postgresql one has to create the URI.

Reply:
No, if you use uri='mongodb://localhost:27017. 
if you are not including username and password (like in mongodb://localhost:27017), no encoding is needed.

Here's an enhanced explanation:

Without username and password: mongodb://localhost:27017
With username and password: mongodb://user:pass@localhost:27017

Reply:
Sure. Convenience functions might help, but we can always add them to help someone not have to remember/know the structure.